Change the Column span prop so it behaves more like Bootstrap

jtanner opened the pull request:
This fixes a problem where the spans were defaulted to 0 when only one
or two of the spans were provided as props which caused the hidden-*
class to be applied causing surprising behavior.

Before:
`<Column span={ large:6, medium:6, small:6 }>`

After:
`<Column span={small:6}>`
This will result in a span of `{ large:6, medium:6, small:6 }`

jmophoto commented:
So, this is what we had before, but with Tommy's changes we changed to follow Bootstraps pattern of only declaring the first property and the rest will follow. So a `small` of 6 will result in a `medium` and `large` of six also. 

I like this way better, though. Caused a lot of confusion for me wondering why widths that were not implicitly set were being `auto-magically` done.

paniclater commented:
I also prefer this way -- but there is definitely something to be said for following bootstrap's convention esp. now that we're using their css.

cerinman commented:
Yeah, I was following the Bootstrap convention.  I lean towards leaving it the way it is.  I think it is more confusing this way as it doesn't match normal css short hand convention but I will bow to the masses if everyone hates it.

jtanner commented on src/components/grid/Column.js:
```diff
@@ -15,30 +15,34 @@ const Column = React.createClass({
     return {
       offset: {},
-      span: { large: 12, medium: 12, small: 12 }
+      span: { large: 12 }
```
^ to give a hint that not all are required

jtanner commented:
After looking at this some more I think the problem lies with [setting a default of 0](https://github.com/mxenabled/mx-react-components/blob/30849faacd85d012f371efcb69a77a9a558158bd/src/components/grid/Column.js#L24) and then [adding the `hidden-*` class](https://github.com/mxenabled/mx-react-components/blob/30849faacd85d012f371efcb69a77a9a558158bd/src/components/grid/Column.js#L30).

I've amended my commit to reflect this new finding.

It looks like @jmophoto already fixed the offsets [here](https://github.com/mxenabled/mx-react-components/commit/30849faacd85d012f371efcb69a77a9a558158bd#diff-e08bcf18e26bf986765255f1a2c623be).

jtanner commented:
@jmophoto @paniclater @cerinman this is ready for another look.

paniclater commented:
:shipit:

jtanner commented on demo/app.js:
```diff
@@ -452,19 +452,23 @@ const Demo = React.createClass({
               </Column>
               <Column
-                span={{ large: 4, medium: 9 }}
+                span={{ large: 4, medium: 6, small: 0 }}
```
I used `small: 0` to verify that the `hidden-sm` class is used.

jtanner commented:
@paniclater @cerinman can I get another look at this? I did some testing with the demo page and found that `if (small) { ...` was not inserting the `hidden-sm` class. I fixed it and also updated the demo page.

cerinman commented:
:shipit:
